File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/Stopped.java
package net.cyclestreets.liveride;

import net.cyclestreets.routing.Journey;

import org.osmdroid.util.GeoPoint;

import android.content.Context;

final class Stopped extends LiveRideState
{
  Stopped(final Context context) {
    super(context, null);
    cancelNotification();
  }

  @Override
  public LiveRideState update(Journey journey, GeoPoint whereIam, int accuracy) {
    return this;
  }

  @Override
  public boolean isStopped() { return true; }
  @Override
  public boolean arePedalling() { return false; }
}

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/Stopped.kt
package net.cyclestreets.liveride

import android.content.Context
import net.cyclestreets.routing.Journey
import org.osmdroid.util.GeoPoint

internal class Stopped(context: Context) : LiveRideState(context, null) {

    init {
        cancelNotification()
    }

    override fun update(journey: Journey, myLocation: GeoPoint, accuracy: Int): LiveRideState {
        return this
    }

    override fun isStopped(): Boolean { return true }
    override fun arePedalling(): Boolean { return false }
}
